### DISCUSSION TITLE
Can't use non-fqcn service id when using ServiceSubscriberInterface. Why?

### DISCUSSION THREAD

**Discussion by ghost** [symfony/symfony#47344] [Q&A] [answered]
When using ServiceSubscriberInterface I can't use the non-fqcn service id. Why? What's the limitation?

The following:
```php
    public static function getSubscribedServices()
    {
        return [
            'app.service.my_service'
        ];
    }
```
throws:
`"App\MyController::getSubscribedServices()" must return valid PHP types for service "App\MyController" key "0", "app.service.my_service" returned.`

**Comment by MatTheCat** [comment, marked as answer]
From [the interface you mention](https://github.com/symfony/symfony/blob/e1581a0937c7e0c8d8420808ebc60418a63892c4/src/Symfony/Contracts/Service/ServiceSubscriberInterface.php#L34-L35):

> Returns an array of service types

`app.service.my_service` is a service name, not type.

You’ll have to tag your subscriber yourself: https://symfony.com/doc/current/service_container/service_subscribers_locators.html#aliased-services
